packages/sdk-js/src/DidHelpers/verificationMethod.spec.ts

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright (c) 2018-2024, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import type { DidDocument, KiltKeyringPair } from '@kiltprotocol/types'
+import { Crypto } from '@kiltprotocol/utils'
+import {
+  ApiMocks,
+  createLocalDemoFullDidFromKeypair,
+} from '../../../../tests/testUtils/index.js'
+import { ConfigService } from '../index.js'
+import { transactInternal } from './transact.js'
+import {
+  removeVerificationMethod,
+  setVerificationMethod,
+} from './verificationMethod.js'
+
+jest.mock('./transact.js')
+
+const mockedTransact = jest.mocked(transactInternal)
+const mockedApi = ApiMocks.createAugmentedApi()
+
+let didDocument: DidDocument
+let keypair: KiltKeyringPair
+beforeAll(async () => {
+  ConfigService.set({ api: mockedApi })
+
+  keypair = Crypto.makeKeypairFromUri('//Alice')
+  const { id, verificationMethod, authentication } =
+    await createLocalDemoFullDidFromKeypair(keypair, {
+      verificationRelationships: new Set(['assertionMethod']),
+    })
+  didDocument = {
+    id,
+    authentication,
+    assertionMethod: authentication,
+    verificationMethod: verificationMethod?.filter(
+      (vm) => vm.id === authentication![0]
+    ),
+  }
+})
+
+describe('signing keys', () => {
+  it('creates a set VM tx', async () => {
+    setVerificationMethod({
+      didDocument,
+      api: mockedApi,
+      submitter: keypair,
+      signers: [keypair],
+      publicKey: keypair,
+      relationship: 'assertionMethod',
+    })
+
+    expect(mockedTransact).toHaveBeenLastCalledWith(
+      expect.objectContaining<Partial<Parameters<typeof transactInternal>[0]>>({
+        callFactory: expect.any(Function),
+        expectedEvents: expect.arrayContaining([
+          {
+            section: 'did',
+            method: 'DidUpdated',
+          },
+        ]),
+        didDocument,
+        api: mockedApi,
+        submitter: keypair,
+        signers: [keypair],
+      })
+    )
+    expect(
+      await mockedTransact.mock.lastCall?.[0]
+        .callFactory()
+        .then((f) => f.toHuman())
+    ).toMatchObject({
+      method: {
+        section: 'did',
+        method: 'setAttestationKey',
+        args: { new_key: { Ed25519: Crypto.u8aToHex(keypair.publicKey) } },
+      },
+    })
+  })
+
+  it('creates a remove VM tx', async () => {
+    didDocument.assertionMethod = didDocument.authentication
+    removeVerificationMethod({
+      didDocument,
+      api: mockedApi,
+      submitter: keypair,
+      signers: [keypair],
+      verificationMethodId: didDocument.assertionMethod![0],
```

Comment:
I tried this and it seems not to work. I am getting the result back of my previous operation.

`     resultSetAssertion.asConfirmed.didDocument.assertionMethod![0],`

However, it gives me the following as an example: `did:kilt:4oZJFxJJGqJJnoNsajpCQmpWg5RrbSHCYQR3bw1vCeafkHto#0xbfa824220a126fdb757036ac3718b4271976dab2e0555a4a23be4eab6257ee19`

I suspect it only wants the following:`#0xbfa824220a126fdb757036ac3718b4271976dab2e0555a4a23be4eab6257ee19`

Reply:
It does work exactly like this in the integration tests though, including assertions that the did no longer has a verification method afterwards. Would be interesting to see the code you used to make sure we don't have a hidden bug here

Reply:
Yes, I was submitting the original did document in the options and not the updated document